### ISSUE TITLE
Idea: use Markov chains and charts library to generate new charts!

### ISSUE THREAD

**Issue by Janiczek** [open-chords-charts/chart-dsl#1]
To get this out of the way: thanks for the talk at ElmEurope 2017 - I really enjoyed it :)

It could be nice experiment to take the existing charts library, normalize it into Nashville number system or something similar (to get all the charts into the same key) and teach a Markov chain with it, and **see what kind of music does it generate!**

If you had a database of charts, I would volunteer to get my Markov chain library up to speed (read: 0.18 or 0.19 :) ) and try to do something with your data!

**Comment by cbenz** [comment]
Hi @Janiczek 

Thanks for your idea :-) I discussed about it with Evan too, during the Elm Europe conference.

About the *Nashville number system*, perhaps what you mean is to store the chords using an absolute number (like `II V I` progressions), which would avoid storing the global chart key in the `Chart` record.

About *Markov chains* we could either generate new charts, either music. The former implies having a database of charts, and the latter a database of melodies.

Basically the roadmap is to finish the basic features of the chart editor, have a decent(ralized) backend, and populate the database with charts.

After that I'm interested too in chords progressions generation based on Markov chains, as an experiment.

Is your library published?

**Comment by cbenz** [comment]
Here is it, but as you said, not up-to-date with the latest Elm version: http://package.elm-lang.org/packages/janiczek/elm-markov/latest

(Using Markov chains is not a priority for me now, there is no urgency to update the library)
